chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PendingSurveillanceManagerImpl.java

```diff
@@ -0,0 +1,728 @@
+        if (pendingSurv.getDeleted()) {
+            ObjectMissingValidationException alreadyDeletedEx = new ObjectMissingValidationException();
+            alreadyDeletedEx.getErrorMessages()
+            .add("This pending surveillance has already been confirmed or rejected by another user.");
+            alreadyDeletedEx.setObjectId(pendingSurv.getId().toString());
+            alreadyDeletedEx.setStartDate(pendingSurv.getStartDate());
+            alreadyDeletedEx.setEndDate(pendingSurv.getEndDate());
+
+            try {
+                UserDTO lastModifiedUser = userDao.getById(pendingSurv.getLastModifiedUser());
+                if (lastModifiedUser != null) {
+                    Contact contact = new Contact();
+                    contact.setFullName(lastModifiedUser.getFullName());
+                    contact.setFriendlyName(lastModifiedUser.getFriendlyName());
+                    contact.setEmail(lastModifiedUser.getEmail());
+                    contact.setPhoneNumber(lastModifiedUser.getPhoneNumber());
+                    contact.setTitle(lastModifiedUser.getTitle());
+                    alreadyDeletedEx.setContact(contact);
+                } else {
+                    alreadyDeletedEx.setContact(null);
+                }
+            } catch (final UserRetrievalException ex) {
+                alreadyDeletedEx.setContact(null);
+            }
+            throw alreadyDeletedEx;
+        }
+        //If pendingSurv were null, we would have gotten an NPE by this point
+        //return pendingSurv != null;
+        return true;
+    }
+
+    private Integer getSurveillanceRecordThreshold() {
+        String surveillanceThresholdToProcessAsJobStr = env.getProperty("surveillanceThresholdToProcessAsJob").trim();
+        Integer surveillanceThresholdToProcessAsJob = SURV_THRESHOLD_DEFAULT;
+        try {
+            surveillanceThresholdToProcessAsJob = Integer.parseInt(surveillanceThresholdToProcessAsJobStr);
+        } catch (final NumberFormatException ex) {
+            LOGGER.error(
+                    "Could not format " + surveillanceThresholdToProcessAsJobStr + " as an integer. Defaulting to"
+                            + " 50 instead.");
+        }
+        return surveillanceThresholdToProcessAsJob;
+    }
+
+    private void validate(final Surveillance surveillance) {
+        validator.validate(surveillance, false);
+    }
+
+    private void checkSurveillanceAuthority(final Surveillance surv) throws SurveillanceAuthorityAccessDeniedException {
+        Boolean hasOncAdmin = Util.isUserRoleAdmin();
+        Boolean hasAcbAdmin = Util.isUserRoleAcbAdmin();
+        if (StringUtils.isEmpty(surv.getAuthority())) {
+            // If user has ROLE_ADMIN and ROLE_ACB
+            // return 403
+            if (hasOncAdmin && hasAcbAdmin) {
+                String errorMsg = "Surveillance cannot be created by user having " + Authority.ROLE_ADMIN + " and "
```

Comment:
Why can't one be created by someone who's both ONC and ACB admin? I know it's unlikely that there'd be a real person like that, but why can't they?

Reply:
I think we have/had that same logic for when a user would manually create a surveillance. If they have both roles we don't know what authority to assign. If we are letting them select an authority in the case that they have both then we wouldn't need this logic. I can't remember if we are doing that or not.

Reply:
I took this method directly from the SurveillanceManager.

Reply:
Wherever it came from, I don't think it's right. If we want to implement a business rule that no user who is ROLE_ACB and either of ROLE_ADMIN or ROLE_ONC can create a surveillance then that's fine, but if we do that we'll need to get buy off from the Product Owner.

One thing we have to do with this project is always have the ROLE interaction challenges in the back of our minds, because there's nothing in any of our business rules that restrict any single User from having more than one ROLE

Reply:
After some analysis, this method is not needed based on the updated security.